### PR TITLE
Profile plot annotate title and text. Addresses Issue #1599

### PR DESCRIPTION
**Pull request by git-abhishek**
## PR Summary

With this change we can add plot title and add annotations in ProfilePlot.

https://github.com/yt-project/yt/issues/1599
## PR Checklist

- [x] Code passes flake8 checker
- [x] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: yt/visualization/profile_plotter.py [hunk 1]
```diff
@@ -711,6 +727,64 @@ def _get_field_title(self, field_y, profile):
...
+        >>>  plot = yt.ProfilePlot(my_galaxy, "density", ["temperature"])
+        >>>  plot.annotate_text(1e-26, 1e5, "This is annotated text in the plot area.")
+        >>>  plot.save()
+
+        """
+        for f in list(self.axes.keys()):
+            if self.plots[f].figure is not None and text is not None:
+                self.plots[f].axes.text(xpos, ypos, text,
+                                        fontproperties=self._font_properties,
+                                        **text_kwargs)
```
Comment (ngoldbaum) [inline yt/visualization/profile_plotter.py]: You can delete this if statement if you decorate this function with `@invalidate_plot`.

File: doc/source/visualizing/plots.rst [hunk 2]
```diff
@@ -1099,6 +1099,38 @@ function of radius. The xlabel is set to "Radius", for all plots, and the ylabel
    plot.set_ylabel("x-velocity", "velocity in x direction")
    plot.save()
 
+Annotating plot Title and Area
```
[the remark this answers is not in the document]
Comment (git-abhishek) [reply doc/source/visualizing/plots.rst]: By Area, I meant plot area (plot region) but I think your suggestion is better. Will update with all necessary updates.

**Comment by git-abhishek** [comment]
After your ``field`` suggestion for ``annotate_title``,  I made few other changes to reflect the same. Also, updated ``annotate_text`` to accept ``field`` parameter as well. Let me know, if I missed anything and thank you for reviewing my code!

File: doc/source/visualizing/plots.rst [hunk 3]
```diff
@@ -1099,6 +1099,83 @@ function of radius. The xlabel is set to "Radius", for all plots, and the ylabel
    plot.set_ylabel("x-velocity", "velocity in x direction")
    plot.save()
 
+Adding plot title
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ngoldbaum) [inline doc/source/visualizing/plots.rst]: The underlines here should match the length of the title

File: doc/source/visualizing/plots.rst [hunk 4]
```diff
@@ -1099,6 +1099,83 @@ function of radius. The xlabel is set to "Radius", for all plots, and the ylabel
...
+   plot.annotate_title("Plot Title: Density", "density")
+   plot.annotate_title("Plot Title: Dark Matter Density", "dark_matter_density")
+   plot.save()
+
+Here, ``plot.annotate_title("Plot Title: Density", "density")`` will only set the plot title for the ``"density"``
+field. Thus, allowing us the option to have different plot titles for different fields.
+
+
+Annotating plot with text
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ngoldbaum) [inline doc/source/visualizing/plots.rst]: same here

Comment (git-abhishek) [reply doc/source/visualizing/plots.rst]: Thank you for pointing out! I fixed those now.

**Pull request merged**